File: ios/Approach/Sources/StatisticsLibrary/Protocols/HighestOfStatistic.swift
import Charts
import ModelsLibrary

public protocol HighestOfStatistic: Statistic {
	var highest: Int { get set }
}

extension HighestOfStatistic {
	public static var supportsAggregation: Bool {
		true
	}

	public mutating func aggregate(with: Statistic) {
		guard let with = with as? Self else { return }
		self.highest = max(self.highest, with.highest)
	}

	public var formattedValue: String {
		String(highest)
	}

	public var isEmpty: Bool {
		highest == 0
	}
}

File: android/core/statistics/src/main/java/ca/josephroque/bowlingcompanion/core/statistics/interfaces/HighestOfStatistic.kt
package ca.josephroque.bowlingcompanion.core.statistics.interfaces

import ca.josephroque.bowlingcompanion.core.statistics.Statistic

interface HighestOfStatistic : Statistic {
	var highest: Int

	override val supportsAggregation: Boolean
		get() = true

	override val supportsWidgets: Boolean
		get() = true

	override val formattedValue: String
		get() = highest.toString()

	override val formattedValueDescription: String?
		get() = null

	override val isEmpty: Boolean
		get() = highest == 0

	override fun aggregateWithStatistic(statistic: Statistic) {
		if (statistic is HighestOfStatistic) {
			highest = maxOf(highest, statistic.highest)
		}
	}
}
